src/wallet/wallet.cpp

```diff
@@ -3189,39 +3189,6 @@ const CAddressBookData* CWallet::FindAddressBookEntry(const CTxDestination& dest
     return &address_book_it->second;
 }
 
-bool CWallet::UpgradeWallet(int version, bilingual_str& error)
-{
-    int prev_version = GetVersion();
-    if (version == 0) {
-        WalletLogPrintf("Performing wallet upgrade to %i\n", FEATURE_LATEST);
-        version = FEATURE_LATEST;
-    } else {
-        WalletLogPrintf("Allowing wallet upgrade up to %i\n", version);
-    }
-    if (version < prev_version) {
-        error = strprintf(_("Cannot downgrade wallet from version %i to version %i. Wallet version unchanged."), prev_version, version);
-        return false;
-    }
-
-    LOCK(cs_wallet);
-
-    // Do not upgrade versions to any version between HD_SPLIT and FEATURE_PRE_SPLIT_KEYPOOL unless already supporting HD_SPLIT
-    if (!CanSupportFeature(FEATURE_HD_SPLIT) && version >= FEATURE_HD_SPLIT && version < FEATURE_PRE_SPLIT_KEYPOOL) {
-        error = strprintf(_("Cannot upgrade a non HD split wallet from version %i to version %i without upgrading to support pre-split keypool. Please use version %i or no version specified."), prev_version, version, FEATURE_PRE_SPLIT_KEYPOOL);
-        return false;
-    }
-
-    // Permanently upgrade to the version
-    SetMinVersion(GetClosestWalletFeature(version));
```

Comment:
There is another instance of `SetMinVersion` in this file . Since `Createwallet` will create a wallet with the newest version , that use is also a dead code . Might as well remove this function too.

Reply:
I don't think we need it either, for the history we could still browse back this PR in the repo.

Reply:
If we simply delete these functions, there will be issues with the `getwalletinfo` and `migratewallet` RPCs, as they call `CanSupportFeature`, which calls `IsFeatureSupported`, which checks the `CWallet::nWalletVersion` attribute.

Those functions can potentially be removed, but the following code snippets need to be addressed first:: https://github.com/bitcoin/bitcoin/blob/6a13a6106e3c1ebe95ba6430184d6260a7b942bd/src/wallet/rpc/wallet.cpp#L92-L94 and https://github.com/bitcoin/bitcoin/blob/6a13a6106e3c1ebe95ba6430184d6260a7b942bd/src/wallet/scriptpubkeyman.cpp#L630-L632

Reply:
@Prabhat1308 @pablomartin4btc thanks for the suggestion. Done in https://github.com/bitcoin/bitcoin/pull/32977.